assets/js/components/record_details/clc_sync.js

```diff
@@ -0,0 +1,287 @@
+import React, { Component } from "react";
+import PropTypes from "prop-types";
+import {
+  Button,
+  Grid,
+  Icon,
+  Message,
+  Checkbox,
+  Divider,
+  Header,
+  Popup,
+} from "semantic-ui-react";
+import { i18next } from "@translations/invenio_app_rdm/i18next";
+import { withCancel, http } from "react-invenio-forms";
+import { DateTime } from "luxon";
+
+export class CLCSync extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      clcSyncRecord: null,
+      loading: false,
+      error: null,
+      synced: null,
+      autoSync: null,
+      showSuccess: false,
+      error: null,
+    };
+  }
+
+  fetchSyncRecord = async () => {
+    const { record } = this.props;
+    try {
+      this.cancellableRequest = withCancel(http.get(`/api/clc/${record.id}`));
+      const response = await this.cancellableRequest.promise;
+      const data = response.data;
+      if (data) {
+        this.setState({
+          synced: true,
+          clcSyncRecord: data,
+          autoSync: data.auto_sync,
+        });
+      }
+    } catch (error) {
+      this.setState({ synced: false });
+    }
+  };
+
+  componentWillUnmount() {
+    if (this.cancellableRequest) {
+      this.cancellableRequest.cancel();
+    }
+  }
+
+  componentDidMount() {
+    this.fetchSyncRecord();
+  }
+
+  syncWithCLC = async (payload, existingId = null) => {
+    if (existingId) {
+      this.cancellableRequest = withCancel(
+        http.put(`/api/clc/${existingId}`, payload, {
+          headers: {
+            "Content-Type": "application/json",
+            Accept: "application/json",
+          },
+        })
+      );
+    } else {
+      this.cancellableRequest = withCancel(
+        http.post(`/api/clc/`, payload, {
```

Comment:
very nitpicking here: the endpoint could be a config var but I think not in the PR :)